Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/JoiSchemaEnhancer.ts

```diff
@@ -49,23 +62,50 @@ function joiTypeForReferentialProperty(property: ReferentialProperty, propertyMo
 }
 
 /**
- * Returns a Joi schema fragment that specifies the API body element shape
- * corresponding to the given common property.
+ * Returns a Joi schema of the properties of a referenced Common entity, handling issues of parent naming,
+ * and choice/inline common property pull-up
  */
-function joiTypeForCommonProperty(property: CommonProperty, propertyModifier: PropertyModifier) {
+function schemaFromPropertiesOfCommon(property: CommonProperty, propertyModifier: PropertyModifier) {
```

Comment:
This block of three functions re-does common property handling, mostly adding common collection support.